File: clients/js/healthApi.js
// Simple JS client for the Health Data Science API
export class HealthApi {
  constructor(baseUrl, apiKey = null, token = null) {
    this.baseUrl = baseUrl.replace(/\/$/, '');
    this.headers = {};
    if (apiKey) this.headers['X-API-Key'] = apiKey;
    if (token) this.headers['Authorization'] = `Bearer ${token}`;
  }

  async health() {
    const res = await fetch(`${this.baseUrl}/v1/health/live`, { headers: this.headers });
    if (!res.ok) throw new Error(await res.text());
    return res.json();
  }
}

File: clients/R/health_api.R
# Simple R wrapper for the Health Data Science API

health_api <- function(base_url, api_key = NULL, token = NULL) {
  headers <- list()
  if (!is.null(api_key)) headers$`X-API-Key` <- api_key
  if (!is.null(token)) headers$Authorization <- paste("Bearer", token)
  list(
    health = function(){
      httr::GET(paste0(base_url, "/v1/health/live"), httr::add_headers(.headers = headers))
    }
  )
}
